### PR TITLE
Refactor Win32 file dialog API

### PR DESCRIPTION
**Pull request by AmelBawa-msft**
## Summary of the pull request
- Added support for file dialogs using win32 API:
  - Open file dialog
  ```cs
  using var fileDialog = new WindowOpenFileDialog();

  fileDialog.AddFileType("YAML files", ".yaml", ".yml", ".winget");
  fileDialog.AddFileType("Image files", ".png", ".gif");

  StorageFile? file = await fileDialog.ShowAsync(window);
  ```
  - Open folder dialog
  ```cs
  using var folderDialog = new WindowOpenFolderDialog();
  
  StorageFolder? folder = await folderDialog.ShowAsync(window);
  ```
  - Save file dialog
  ```cs
  using var fileDialog = new WindowSaveFileDialog();
  
  fileDialog.AddFileType("YAML files", ".yaml", ".yml", ".winget");
  fileDialog.AddFileType("Image files", ".png", ".gif");
  
  var filePath = fileDialog.Show(window);
  ```

- This PR also replaces the folder picker usage in Repo and Dev Drive pages preventing the app from crashing in admin mode

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: docs/common/Readme.md [hunk 1]
```diff
@@ -7,3 +7,14 @@ List of common controls that are generic, customizable and reusable from all pag
 
 ## Controls
 - [CloseButton](./CloseButton.md)
+
```
Comment (krschau) [inline docs/common/Readme.md:10]: :heart: docs

**Review by krschau — APPROVED** [review]

**Comment by krschau** [comment]
@joadoumie Let's use this -- doesn't matter in widgets which can't run as admin, but potentially someone could use the AC picker somewhere else.

**Comment by joadoumie** [comment]
> @joadoumie Let's use this -- doesn't matter in widgets which can't run as admin, but potentially someone could use the AC picker somewhere else.

This is just to replace the file picker for #2477?

**Review by dhoehna — APPROVED** [review]

File: common/Windows/FileDialog/WindowFileDialog.cs [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+    }
+
+    /// <summary>
+    /// Class to handle file dialog events.
+    /// </summary>
+    private sealed class WindowFileDialogEvents(WindowFileDialog fileDialog) : IFileDialogEvents
+    {
+        private readonly WindowFileDialog _fileDialog = fileDialog;
+
+        public void OnTypeChange(IFileDialog pfd) => _fileDialog.FileTypeChanged?.Invoke(null, _fileDialog.FileType);
```
[the remark this answers is not in the document]
Comment (AmelBawa-msft) [reply common/Windows/FileDialog/WindowFileDialog.cs:230]: Assuming this is referring to the `pdf` parameter. This parameter is defined in the `IFileDialogEvents::OnTypeChange([in] IFileDialog *pfd);` signature.

Source: [OnTypeChange](https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-ifiledialogevents-ontypechange#parameters)

**Pull request merged**